### DISCUSSION TITLE
Can grain method calls come in and process while grain timer event is firing?

### DISCUSSION THREAD

**Discussion by swegele** [dotnet/orleans#8976] [Q&A] [answered]
Say I had a timer event firing every minute.  Say it fires and is processing...could a grain method be called and now my timer processing code is running at the same time as a grain method?  Would I be guaranteed the "one-at-a-time" queue behavior that grain methods have even though the timer doesn't trigger the normal start of a grain method?

**Comment by swegele** [comment, marked as answer]
Still getting my head wrapped around the terms used here "interleave" and "re-entrancy" but I think[ this topic and related stuff](https://github.com/dotnet/orleans/issues/2574) answeres my question for now and maybe for future enhancements too.

**Reply by ReubenBond** [reply]
Yes, grain timers are currently "interleaving" (aka reeentrant). Here is the PR where we're implementing which act like regular grain calls: https://github.com/dotnet/orleans/pull/8955
